test/jdk/javax/swing/JSpinner/4670051/JSpinnerFieldUnderCursorTest.java

```diff
@@ -0,0 +1,302 @@
+                downValue = spinnerValue;
+                verifyDayDecrement();
+
+                // Cursor at Month Field
+                pressRightArrowKey();
+                // Increment Month
+                updateSpinnerValue();
+                initValue = spinnerValue;
+                mousePressOnUpButton();
+                updateSpinnerValue();
+                upValue = spinnerValue;
+                verifyMonthIncrement();
+                // Decrement Month
+                updateSpinnerValue();
+                initValue = spinnerValue;
+                mousePressOnDownButton();
+                updateSpinnerValue();
+                downValue = spinnerValue;
+                verifyMonthDecrement();
+
+                // Cursor at Year Field
+                pressRightArrowKey();
+                // Increment Year
+                updateSpinnerValue();
+                initValue = spinnerValue;
+                mousePressOnUpButton();
+                updateSpinnerValue();
+                upValue = spinnerValue;
+                verifyYearIncrement();
+                // Decrement Year
+                updateSpinnerValue();
+                initValue = spinnerValue;
+                mousePressOnDownButton();
+                updateSpinnerValue();
+                downValue = spinnerValue;
+                verifyYearDecrement();
+
+                if (passed) {
+                    System.out.println("Test Passed");
+                } else {
+                    throw new RuntimeException("Test Failed as one or more cases failed");
+                }
+            } finally {
+                SwingUtilities.invokeAndWait(JSpinnerFieldUnderCursorTest::disposeFrame);
+            }
+        }
+    }
+
+    private static void updateSpinnerValue() throws Exception {
+        SwingUtilities.invokeAndWait(() -> spinnerValue = (Date) spinner.getValue());
+    }
+
+
+    public static void pressRightArrowKey() {
+        robot.keyPress(KeyEvent.VK_RIGHT);
+        robot.keyRelease(KeyEvent.VK_RIGHT);
+    }
+
+
+    public static void mousePressOnUpButton() {
+        robot.mouseMove(spinnerUpButtonCenter.x, spinnerUpButtonCenter.y);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+    }
+
+    public static void mousePressOnDownButton() {
+        robot.mouseMove(spinnerDownButtonCenter.x, spinnerDownButtonCenter.y);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+    }
+
+    public static void main(String[] s) throws Exception {
+        runTest();
+    }
+
+    public static boolean compareDates(Calendar d1, Calendar d2) {
+        return (d1.get(Calendar.DATE) == d2.get(Calendar.DATE)) &&
+                (d1.get(Calendar.DAY_OF_YEAR) == d2.get(Calendar.DAY_OF_YEAR)) &&
+                (d1.get(Calendar.DAY_OF_MONTH) == d2.get(Calendar.DAY_OF_MONTH));
+    }
```

Comment:
`DATE` and `DAY_OF_MONTH` return the same value.

Does it make sense to compare `MONTH` and `YEAR`? You never compare that the months or years are equal, the former, however, should be covered by `DAY_OF_YEAR`.